### DISCUSSION TITLE
Privacy/Cloud Storage

### DISCUSSION THREAD

**Discussion by alyoung1221** [allure-framework/allure2#2341] [Questions & Support]
Does [Allure Reports](https://allurereport.org/) store the generated reports or any data used to generate the reports i.e. Allure parameters in the cloud? I am using [allure-playwright](https://www.npmjs.com/package/allure-playwright) installed via NPM.

**Comment by baev** [comment]
Allure itself doesn't require you to store data on the cloud and allows you to fully use its functionality locally or keep the reports on CI. 

However, there are a few options you may consider:

1. [Allure TestOps](https://qameta.io/) — a solution the Allure Team provides, a bit more than just a reporting storage. Brings extended analytics, real-time reporting, selective test runs, test management features and much more. Available both cloud & self-hosted installations.
2. You can deploy a simple Docker-based reporting service by yourself, e.g. https://github.com/fescobar/allure-docker-service
3. You can upload reports into your own S3 storage. All you need to do is to generate the report and upload all the files in `allure-report` folder to a bucket. The only requirement is to have file storage with path-style access enabled. You can do it by using https://github.com/andrcuns/allure-report-publisher. Besides that, depending on the CI, there are probably some integrations available, e.g. for GitHub Actions: 
   a. https://github.com/marketplace/actions/publish-allure-report-to-morph-object-storage
   b. https://github.com/marketplace/actions/allure-html-reporter-aws-s3-website

**Reply by alyoung1221** [reply]
When generating HTML reports and running a local web server with Allure Report command-line tool using `npx allure serve <dir>`,  is any data sent or stored in the cloud?

**Reply by epszaw** [reply]
No, it's not. All data you generate is on your machine.

**Reply by alyoung1221** [reply]
Does Allure use any external libraries/dependencies that sends or stores data in the cloud/databases?
